docs/source/containers.rst

```diff
@@ -1,34 +1,300 @@
 
-.. autoclass:: happi.OphydItem
-    :members:
+* Identifying how to instantiate the object it represents (by way of class
+  name, args, and keyword arguments).
+* Storing pertinent and structured metadata for the given instance.
 
-These inherit from the most generic information that can be included:
+Containers are created by subclassing from the built-in :class:`.HappiItem`
+class.  The metadata associated with the instance is broken up into fields or
+"entries" of type :class:`.EntryInfo`.
 
-.. autoclass:: happi.HappiItem
-    :members:
+EntryInfo
+^^^^^^^^^
 
-Extending Containers
---------------------
+In order to regulate and template the information that gets entered into the
+Happi database, we use the concept of containers. This allows a developer to
+specify fields that are essential to every instance of a specific type.
 
-Happi allow users to extend the containers list via Python
-`entry points <https://packaging.python.org/specifications/entry-points/>`_
-specified by the **happi.containers** keyword. An example entry points can be
-found `here <https://github.com/pcdshub/pcdsdevices/blob/master/setup.py>`_.
+These fields are specified using an instance of :class:`.EntryInfo`.
+`.EntryInfo` provides several primary features:
 
-`HappiRegistry` takes care of loading the entry points and making them
-available throughout the library.
+* Mark a field as required (or optional)
+* Add default values when unspecified
+* Enforce - or validate - a certain format for the field
 
-Loading Containers
-------------------
+.. autoclass:: happi.EntryInfo
+   :members:
 
-.. autofunction:: happi.from_container
+HappiItem
+^^^^^^^^^
 
-.. autofunction:: happi.load_devices
+In order to ensure that information is entered into the database in an
+organized fashion, the client will only accept classes that inherit from
+:class:`.HappiItem`. Each item will have the key information represented as
+class attributes, available to be manipulated like any other regular property
+
+Editing the information for a container is a simple as:
+
+.. ipython:: python
+
+    from happi import HappiItem
+
+    item = HappiItem(name='my_device')
+
+    item.name = 'new_name'
+
+
+.. note::
+
+    :class:`happi.Device` class is **deprecated** due to ambiguous name,
+    conflicting with :class:`ophyd.Device`.
+
+
+Example Container
+^^^^^^^^^^^^^^^^^
+
+In order to show the flexibility of the :class:`.EntryInfo` class, we'll put
+together a new example container. The class can be invoked in the same way you
+would usually handle class inheritance, the only difference is that you specify
+class attributes as EntryInfo objects:
+
+.. ipython:: python
+
+    import re
+    from happi import HappiItem, EntryInfo
+
+    class MyDevice(HappiItem):
+        """My new device, with a known model number."""
+        model_no      = EntryInfo('Model Number of Device', optional=False)
+        count         = EntryInfo('Count of Device', enforce=int, default=0)
```

Comment:
We should further purge the `Device` keyword to avoid confusion/bad habits:
```suggestion
    class MyItem(HappiItem):
        """My new item, with a known model number."""
        model_no      = EntryInfo('Model Number of Item', optional=False)
        count         = EntryInfo('Count of Item', enforce=int, default=0)
```